src/message/connection-endpoint.js

```diff
@@ -376,6 +380,12 @@ ConnectionEndpoint.prototype._registerAuthenticatedSocket  = function( socketWra
 		socketWrapper.sendMessage( C.TOPIC.AUTH, C.ACTIONS.ACK, [ messageBuilder.typed( userData.clientData ) ] );
 	}
 
+	if( userData.username != 'OPEN' ) {
+		var msg = messageBuilder.getMsg( C.TOPIC.PRESENCE, C.ACTIONS.PRESENCE_ADD, [ messageBuilder.typed( userData.username ) ] );
+		debugger;
+		setTimeout( this.onMessage( socketWrapper, msg ), 0);
```

Comment:
We might also need to actually call the method directly on the presense, i been thinking about this approach and it can easily allow users to trigger the same event using that topic and message which would break the state of the system. See deepstream.io comment